### PR TITLE
Replace JWT library for signing tokens

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
* Extending `Claims` with "Registered claims" from https://www.rfc-editor.org/rfc/rfc7519#section-4.1
* Defining issuer for easier key detection on server-side
* Adding `ExpiresAt` but also keeping TTL for backwards compatibility

**Why?**
To align with upcoming server changes

**How did you test it?**
Unit tests

**Potential risks**

### CODE REVIEW

File: internal/common/auth/service_wrapper.go [hunk 1]
```diff
@@ -30,7 +31,8 @@ import (
 )
 
 const (
-	jwtHeaderName = "cadence-authorization"
+	jwtHeaderName     = "cadence-authorization"
+	JWTInternalIssuer = "internal-jwt"
```
Comment (3vilhamster) [inline internal/common/auth/service_wrapper.go:35]: I propose separating internal constants from external.

https://github.com/uber-go/guide/blob/master/style.md#group-similar-declarations

File: internal/common/auth/service_wrapper.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ package auth
 
 import (
 	"context"
+	"github.com/golang-jwt/jwt/v5"
```
Comment (3vilhamster) [inline internal/common/auth/service_wrapper.go:25]: separate std imports from others.

Unrelated @Groxx Should we onboard client to gci too?

File: internal/jwt_authorization.go [hunk 3]
```diff
@@ -40,24 +40,28 @@ func NewAdminJwtAuthorizationProvider(privateKey []byte) auth.AuthorizationProvi
...
-	}
 	key, err := util.LoadRSAPrivateKey(j.PrivateKey)
 	if err != nil {
 		return nil, err
 	}
-	signer, err := jwt.NewSignerRS(jwt.RS256, key)
-	if err != nil {
-		return nil, err
+
+	ttl := int64(60 * 10)
```
Comment (3vilhamster) [inline internal/jwt_authorization.go:48]: why it is hardcoded? Should it be configurable?

File: internal/jwt_authorization_test.go [hunk 4]
```diff
@@ -21,60 +21,41 @@
 package internal
 
 import (
-	"encoding/json"
-	"io/ioutil"
+	"github.com/stretchr/testify/assert"
```
Comment (3vilhamster) [inline internal/jwt_authorization_test.go:24]: goimports

**Review by 3vilhamster — COMMENTED** [review]
Have some questions, but overall looks ok.

File: internal/jwt_authorization_test.go (hunk 4 above)
Comment (mantas-sidlauskas) [reply internal/jwt_authorization_test.go:24]: done

File: internal/common/auth/service_wrapper.go (hunk 1 above)
Comment (mantas-sidlauskas) [reply internal/common/auth/service_wrapper.go:35]: thanks, moving this const to another package, this doesn't belong here.

File: internal/jwt_authorization.go (hunk 3 above)
Comment (mantas-sidlauskas) [reply internal/jwt_authorization.go:50]: No sure why this was hard-coded during the [initial implementation](https://github.com/uber-go/cadence-client/pull/1116) but my take is this: this ttl is for internal/self-signed tokens. There are no external calls to acquire a token, there's no caching involved, they are issued for every request, in other words, they are cheap and should be short lived.

**Review by 3vilhamster — APPROVED** [review]
LGTM

**Pull request merged**